### PR TITLE
Fix camera mute

### PR DESCRIPTION
**Pull request by xp-vit**
On some devices even if playSounds = "false" the shutter sound is still played. This pull request handles this issue on those devices where it's possible to mute programmatically:  
    `if (info.canDisableShutterSound)`

### CODE REVIEW

**Comment by xp-vit** [comment]
@natario1 Can you please restart build on Travis? I don't see the log and can't investigate why it's failed.

**Comment by natario1** [comment]
I did now! Thanks for the contribution, a couple issues I see:

- for this to work, CameraView class must pass the playsounds to the controller, you missed this step 🙂
- setPlaySounds (thinking about the controller) should be similar to the other set* methods. You should use the schedule() method, it posts to a queue and ensures that the camera is available
- applyPlaySounds should be mergePlaySounds and return a boolean, like every other does. Take a look
- after you have fixed these, it would be great if you could add some tests. There are tests for other parameters which you could learn from.

This said, the feature is cool, I didn't know about it.

File: cameraview/src/main/java/com/otaliastudios/cameraview/Camera1.java [hunk 1]
```diff
@@ -366,6 +369,20 @@ private boolean mergeHdr(Camera.Parameters params, Hdr oldHdr) {
         return false;
     }
 
+    @TargetApi(17)
+    private boolean mergePlaySound(boolean oldPlaySound) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+            Camera.CameraInfo info = new Camera.CameraInfo();
+            Camera.getCameraInfo(mCameraId, info);
+            if (info.canDisableShutterSound && isCameraAvailable()) {
```
Comment (natario1) [inline cameraview/src/main/java/com/otaliastudios/cameraview/Camera1.java]: Can you remove isCameraAvailable here? it's already managed by the schedule() method

File: cameraview/src/main/java/com/otaliastudios/cameraview/Camera1.java [hunk 2]
```diff
@@ -366,6 +369,20 @@ private boolean mergeHdr(Camera.Parameters params, Hdr oldHdr) {
...
+                mCamera.enableShutterSound(mPlaySounds);
+                return true;
+            }
+        }
+        mPlaySounds = oldPlaySound;
```
Comment (natario1) [inline cameraview/src/main/java/com/otaliastudios/cameraview/Camera1.java]: You should only revert it to the old value if (mPlaySounds == false). The true value should always be allowed, like...

if (apiLevel) { 
    ....
}
if (mPlaySounds) {
    return true; // always on
}
mPlaySounds = oldPlaySounds;
return false;

File: cameraview/src/main/java/com/otaliastudios/cameraview/Camera1.java [hunk 3]
```diff
@@ -858,6 +879,20 @@ void setVideoMaxSize(long videoMaxSizeInBytes) {
...
 
+    @Override
+    void setPlaySounds(boolean playSounds) {
+        final boolean old = mPlaySounds;
+        mPlaySounds = playSounds;
+        schedule(null, true, new Runnable() {
+            @Override
+            public void run() {
+                if (!mergePlaySound(old)) {
+                    mPlaySounds = old;
```
Comment (natario1) [inline cameraview/src/main/java/com/otaliastudios/cameraview/Camera1.java]: This kind of revert is already managed by mergePlaySounds itself. Just call mergePlaySounds() here without any `if`

**Comment by xp-vit** [comment]
@natario1 Fixed all your comments and added an IntegrationTest for setPlaySounds. Please take a look.

Thanks!

**Comment by natario1** [comment]
Thanks man! For the test to work you should pass mPlaySoundsTask to the schedule method, instead of null. The rest looks great!

**Comment by xp-vit** [comment]
@natario1 done. Thanks a lot for quick replies. Hoping to see this code in release soon :-)

**Comment by natario1** [comment]
I'd like to release a new version but I'll be away from keyboard until the end of month. To release, I need a "bump version" commit that does the following:

- upgrade version number in build.gradle file
- upgrade version number in README file
- updates the CHANGELOG file with a report of the changes until the latest version

If you can open a PR with these changes, we can release a new version soon! Take a look at the commit history

**Pull request merged**